onionshare/common.py

```diff
@@ -18,20 +18,26 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 import base64
+import binascii
 import hashlib
 import inspect
 import os
 import platform
 import random
+import re
 import socket
 import sys
 import tempfile
 import threading
 import time
+from flask import abort
```

Comment:
I think I'd be OK with these functions being inside onionshare/web/share_mode.py so we don't have to import flask here, in other words keeping all this stuff specific to the Share Mode in the Web module. It looks like you are importing `flask.abort` already over there anyway.

We use 'Common' mainly for stuff that is used in both CLI and GUI mode, but Web is already common to both too and has all the flask logic.

Just my 2c though, will let @micahflee do a proper review :)

Thanks for contributing this, looking forward to testing soon!